idlookup/idlookup.py

```diff
@@ -0,0 +1,76 @@
+import asyncio
+import os
+import re
+import urllib.parse
+
+from redbot.core import checks, commands, data_manager, Config
+
+import discord
+import tsutils
+
+try:
+    import re2 as re
+except ImportError:
+    try:
+        import regex as re
+    except ImportError:
+        import re
+
+class Idlookup(commands.Cog):
+    """Monster Name from ID"""
+    def __init__(self, bot):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=717432306)
+
+    @commands.Cog.listener('on_message')
+    async def on_message(self, message):
+        channel = message.channel
+        content = message.content
+        if await self.config.channel(channel).enabled():
+            if message.guild is None: #dms
+                return
+            if message.author == self.bot.user: #dont reply to self
+                return
+            if await self.is_command(message): #skip commands
+                return
+            dgcog = self.bot.get_cog("Dadguide")
+            if dgcog is None:
+                await channel.send(inline("Error: Dadguide Cog not loaded.  Please alert a bot owner."))
+                return
+            if re.search(r'\b\d\d\d[ -]?\d\d\d[ -]?\d\d\d\b', content): #friend code
+                return
+            if re.search(r'\d{3,4}', content):
+                matches = re.findall(r'\d{3,4}',content)
+                ret = ""
+                for i in matches:
+                    m = await dgcog.find_monster(i, message.author.id)
+                    if not m: #monster not found
+                        continue
+                    ret+="[{}] {}\n".format(i, m.name_en)
+                await channel.send(ret)
+                return
+        else:
+            return
+
+    @commands.command()
+    @checks.admin_or_permissions(administrator=True)
+    async def idlenable(self, ctx, *channels: discord.TextChannel):
+        """Enable idlookup in channel"""
+        self.config.register_channel(enabled=True)
```

Comment:
The way to set a config object is `self.config.channel(channel).enabled.set(True)`.  If you want this to take multiple channels (from the *channels in the args) you need to loop through them and do this for each channel.